lib/ansible/modules/windows/win_data_deduplication.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: win_data_deduplication
+version_added: "2.9"
+short_description: Module to enable Data Deduplication on a volume.
+description:
+- This module can be used to enable Data Deduplication on a Windows volume.
+- You have to have the FS-Data-Deduplication feature installed.
+- This module doesn't support check_mode due to a lack of WhatIf support.
```

Comment:
The lack of `-WhatIf` shouldn't stop you from implementing check mode. Just knowing whether Ansible will report a change when in check mode is great, we don't need `-WhatIf` on the set cmdlets to know that.